packages/opentelemetry-metrics/src/BaseObserverMetric.ts

```diff
@@ -56,13 +60,23 @@ export abstract class BaseObserverMetric extends Metric<BoundObserver>
   }
 
   getMetricRecord(): Promise<MetricRecord[]> {
-    const observerResult = new ObserverResult();
-    this._callback(observerResult);
-    observerResult.values.forEach((value, labels) => {
-      const instrument = this.bind(labels);
-      instrument.update(value);
+    return new Promise((resolve, reject) => {
+      const observerResult = new ObserverResult();
+      Promise.resolve()
+        .then(() => {
+          return this._callback(observerResult) || Promise.resolve();
+        })
+        .then(() => {
+          observerResult.values.forEach((value, labels) => {
+            const instrument = this.bind(labels);
+            instrument.update(value);
+          });
+          super.getMetricRecord().then(resolve, reject);
```

Comment:
This odd promise chain seems like it would be vastly simpler and more readable if using async/await